### ISSUE TITLE
Assertion fails in some instances

### ISSUE THREAD

**Issue by arpj-rebola** [marijnheule/drat-trim#25]
The proof generated by CaDiCaL on the instance 'baseballcover15with25' from the SAT Competition 2020 fails to check in the current commit. That same proof is checked correctly with commit #d13f761fbdacd052429f14421f95a7e8cd75deb1

**Comment by arpj-rebola** [comment]
[test.log](https://github.com/marijnheule/drat-trim/files/6068947/test.log)

**Comment by krobelus** [comment]
[smaller reproducer](https://github.com/marijnheule/drat-trim/files/6072922/repro.tar.gz) - the proof is not correct but it's enough to  trigger the issue. 
Maybe 39e2006fe77e0033c9e73cb87d17b6d3b508a5f7 needs to be reverted - we somehow get dependencies on later lemmas..

**Comment by krobelus** [comment]
(Fixed by d81931da540024846d528e930731401d7c9bf278)

Minimal reproducer:

```
p cnf 4 6
 4  0
 1  2  0
 1 -2  0
-1 -3  0
 2  3  0
-4 -2 -1 0
```

```
-1 3 0
 4 0
 1 0
```

Since 39e2006, forward check sets the reason for 4 to the second lemma (instead of the first clause in the formula).

Lemma `-1 3 0` has RUP. The conflict graph contains 4, hence the wrong dependency on the second lemma.

**Comment by marijnheule** [comment]
I am not sure what caused this, but the current commit works well on the example.
